From a4da490f75e76e6b633a6b85463ecaf3a26e862c Mon Sep 17 00:00:00 2001
From: Mingliang Liu <liuml07@apache.org>
Date: Mon, 12 Dec 2016 11:58:11 -0800
Subject: [PATCH 2231/2848] HADOOP-13886. s3guard:
 ITestS3AFileOperationCost.testFakeDirectoryDeletion
 failure. Contributed by Mingliang Liu

(cherry picked from commit b2c4f8cfe181d050e54ac29c08d5c0ac888dd4d0)

Conflicts:
	hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFileOperationCost.java

Change-Id: I46115a497ac6bd347c25503cc2b830e1430af9b3
---
 .../hadoop/fs/s3a/ITestS3AFileOperationCost.java   |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFileOperationCost.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFileOperationCost.java
index a9b3396..4569d0e 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFileOperationCost.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AFileOperationCost.java
@@ -34,6 +34,7 @@
 import static org.apache.hadoop.fs.s3a.Statistic.*;
 import static org.apache.hadoop.fs.s3a.S3ATestUtils.*;
 import static org.apache.hadoop.fs.s3a.S3ATestUtils.MetricDiff;
+import static org.junit.Assume.assumeFalse;
 
 /**
  * Use metrics to assert about the cost of file status queries.
@@ -196,6 +197,13 @@ public void testFakeDirectoryDeletion() throws Throwable {
         + "In S3, rename deletes any fake directories as a part of "
         + "clean up activity");
     S3AFileSystem fs = getFileSystem();
+
+    // As this test uses the s3 metrics to count the number of fake directory
+    // operations, it depends on side effects happening internally. With
+    // metadata store enabled, it is brittle to change. We disable this test
+    // before the internal behavior w/ or w/o metadata store.
+    assumeFalse(fs.isMetadataStoreConfigured());
+
     Path srcBaseDir = path("src");
     mkdirs(srcBaseDir);
     MetricDiff deleteRequests =
-- 
1.7.9.5

